src/sentry/integrations/slack/service.py

```diff
@@ -427,21 +460,39 @@ def send_message_to_slack_channel(
         """Execution of send_notification_as_slack."""
 
         client = SlackSdkClient(integration_id=integration_id)
-        try:
-            client.chat_postMessage(
-                blocks=str(payload.get("blocks", "")),
-                text=str(payload.get("text", "")),
-                channel=str(payload.get("channel", "")),
-                unfurl_links=False,
-                unfurl_media=False,
-                callback_id=str(payload.get("callback_id", "")),
-            )
-            metrics.incr(SLACK_NOTIFY_RECIPIENT_SUCCESS_DATADOG_METRIC, sample_rate=1.0)
-        except SlackApiError as e:
-            extra = {"error": str(e), **log_params}
-            self._logger.info(log_error_message, extra=extra)
-            metrics.incr(
-                SLACK_NOTIFY_RECIPIENT_FAILURE_DATADOG_METRIC,
-                sample_rate=1.0,
-                tags={"ok": e.response.get("ok", False), "status": e.response.status_code},
-            )
+        with MessagingInteractionEvent(
+            interaction_type=MessagingInteractionType.SEND_GENERIC_NOTIFICATION,
```

Comment:
this is generic, can be used for activity notifications or sending a slack notification to a team or a user.